pytorch_lightning/trainer/callback_config.py

```diff
@@ -46,13 +46,7 @@ def is_overridden(self, *args):
 
     def configure_checkpoint_callback(self, checkpoint_callback):
         if checkpoint_callback is True:
-            # when no val step is defined, use 'loss' otherwise 'val_loss'
-            train_step_only = not self.is_overridden('validation_step')
-            monitor_key = 'loss' if train_step_only else 'val_loss'
-            checkpoint_callback = ModelCheckpoint(
-                filepath=None,
-                monitor=monitor_key
-            )
+            checkpoint_callback = ModelCheckpoint(filepath=None, monitor='val_loss')
```

Comment:
With this change, what happens if I do not define validation step and only `loss` is available?
Maybe we should have a similar warning as in early stopping:
https://github.com/PyTorchLightning/pytorch-lightning/blob/7b054399c647eac82a9520d4315c40d517fa6c17/pytorch_lightning/callbacks/early_stopping.py#L108-L113